src/main/java/seedu/address/model/client/ClientRebuilder.java

```diff
@@ -0,0 +1,124 @@
+package seedu.address.model.client;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.model.meeting.Meeting;
+import seedu.address.model.product.Product;
+
+/**
+ * This class is used to create a mutable copy of a {@code Client} if a field in {@code Client} needs to be updated
+ * due to the immutability of {@code Client}.
+ */
+public class ClientRebuilder {
+    private Name name;
+    private Phone phone;
+    private Email email;
+    private Address address;
+    private Optional<Birthday> birthday;
+    private Set<Product> products;
+    private List<Meeting> meetings;
+
+    /**
+     * Initializes the ClientBuilder with the data of {@code clientToCopy}.
+     */
+    public ClientRebuilder(Client clientToCopy) {
+        name = clientToCopy.getName();
+        phone = clientToCopy.getPhone();
+        email = clientToCopy.getEmail();
+        address = clientToCopy.getAddress();
+        birthday = clientToCopy.getBirthday();
+        products = new HashSet<>(clientToCopy.getProducts()); // mutable unlike in Client
+        meetings = new ArrayList<>(clientToCopy.getMeetings());
+    }
+
+    /**
+     * Sets the {@code Name} of the {@code Client} that we are building.
+     */
+    public ClientRebuilder withName(Name name) {
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * Sets the {@code Address} of the {@code Client} that we are building.
+     */
+    public ClientRebuilder withAddress(Address address) {
+        this.address = address;
+        return this;
+    }
+
+    /**
+     * Sets the {@code Phone} of the {@code Client} that we are building.
+     */
+    public ClientRebuilder withPhone(Phone phone) {
+        this.phone = phone;
+        return this;
+    }
+
+    /**
+     * Sets the {@code Email} of the {@code Client} that we are building.
+     */
+    public ClientRebuilder withEmail(Email email) {
+        this.email = email;
+        return this;
+    }
+    /**
+     * Sets the {@code Set<Product>} of the {@code Client} that we are building.
+     */
+    public ClientRebuilder withProducts(Set<Product> products) {
+        this.products = products;
+        return this;
+    }
+
```

Comment:
I believe that you can add one more method `withBirthday`?

Reply:
Added in merge commit